packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -235,6 +235,10 @@ export class CurlyComponentDefinition extends ComponentDefinition {
     builder.attrs.static('class', 'ember-view');
   }
 
+  getComponentClass() {
```

Comment:
Definitions cache compiled layout, they can't change class, this is why late bound layouts clone the definition and cache based on template id